tests/Spork/Test/Batch/BatchJobTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Spork\Test\Batch;
+
+class BatchJobTest extends \PHPUnit_Framework_TestCase
+{
+    public function testErrorPropagation()
+    {
+        $manager = new \Spork\ProcessManager();
+        $batch = $manager->createBatchJob(range(1, 5));
+
+        $expectedExitStatus = 20;
+
+        $failingClosure = function ($data) use ($expectedExitStatus) {
+            // Simple condition to simulate only one point of failure
+            if (3 === $data) {
+                exit($expectedExitStatus);
```

Comment:
Exceptions are indeed more sensible use cases. But this way we cannot test the correctness of the exit status.